### DISCUSSION TITLE
Running a while loop only after timeline node

### DISCUSSION THREAD

**Discussion by victoirebeaufils** [jspsych/jsPsych#1976] [Q&A]
Hi! I currently have a timeline in which I'm required to run a while loop. In this loop, I'm adding a tree picture to the timeline, then adding a dot, whose color alternates between black and white, until the user presses a key. The function waitChoice() handles that.

However, as soon as I open my index.html file, the while loop begins. Ideally, I would want it to begin only after the experiment instructions have been read, just as the experiment begins. _**Is there a jspsych built-in function that handles that?**_

The fix that I have right now would be to add the  "on_data_update" parameter to the jspsych init function, and have it run a custom function that uses the DOM to check if an HTML element has been displayed before running the loop.

 
 jsPsych.init({
    timeline: instructionsTimeline,
    display_element: "container",
    on_data_update: _allowForRunTrials_
)}

Here is a code snippet of the while function:

![image](https://user-images.githubusercontent.com/18514535/124613901-b6708500-de41-11eb-8351-9ebab9610519.png)

Thanks in advance for taking the time to read this post!

**Comment by jodeleeuw** [comment]
Hi @victoirebeaufils,

How are you adding and manipulating the display to show the dots and tree? Is it within the context of a plugin, or some other method?
